ledger/ledger.go

```diff
@@ -323,29 +323,63 @@ func (l *Ledger) GetLastCatchpointLabel() string {
 // GetAssetCreatorForRound looks up the asset creator given the numerical asset
 // ID. This is necessary so that we can retrieve the AssetParams from the
 // creator's balance record.
-func (l *Ledger) GetAssetCreatorForRound(rnd basics.Round, assetIdx basics.AssetIndex) (basics.Address, error) {
+func (l *Ledger) GetAssetCreatorForRound(rnd basics.Round, aidx basics.AssetIndex) (creator basics.Address, ok bool, err error) {
 	l.trackerMu.RLock()
 	defer l.trackerMu.RUnlock()
-	return l.accts.getAssetCreatorForRound(rnd, assetIdx)
+	return l.accts.getCreatorForRound(rnd, basics.CreatableIndex(aidx), basics.AssetCreatable)
+}
+
+// GetAppCreatorForRound looks up the application creator given the app ID.
+// This is used to retrieve the AppParams from the creator's balance record.
+func (l *Ledger) GetAppCreatorForRound(rnd basics.Round, aidx basics.AppIndex) (creator basics.Address, ok bool, err error) {
+	l.trackerMu.RLock()
+	defer l.trackerMu.RUnlock()
+	return l.accts.getCreatorForRound(rnd, basics.CreatableIndex(aidx), basics.AppCreatable)
+}
+
+// GetCreatorForRound takes a CreatableIndex and a CreatableType and tries to
+// look up a creator address, setting ok to false if the query succeeded but no
+// creator was found.
+func (l *Ledger) GetCreatorForRound(rnd basics.Round, cidx basics.CreatableIndex, ctype basics.CreatableType) (creator basics.Address, ok bool, err error) {
```

Comment:
Consider adding in `error.go` a new error (maybe with the name `NoCreatorFoundError`) and returning that instead of having an extra `ok bool` return value.